pygments/lexers/maxima.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+    pygments.lexers.maxima
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the computer algebra system Maxima.
+
+    Derived from pygments/lexers/algebra.py.
+"""
+
+import re
+
+from pygments.lexer import RegexLexer, bygroups, words
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, \
+    Number, Punctuation
+
+__all__ = ['MaximaLexer']
+
+class MaximaLexer(RegexLexer):
+    """
+    A `Maxima <http://maxima.sourceforge.net>`_ lexer.
+    Derived from pygments.lexers.MuPADLexer.
+    """
+    name = 'Maxima'
+    aliases = ['maxima', 'macsyma']
+    filenames = ['*.mac', '*.max']
+
+    terminated = r'(?=[ "()\'\n,;`])'  # whitespace or terminating macro characters
+
+    tokens = {
+        'root': [
+            (r'/\*', Comment.Multiline, 'comment'),
+            (r'"(?:[^"\\]|\\.)*"', String),
+            (r'\(|\)|\[|\]|\{|\}', Punctuation),
+            (r'''(?x)\b(?:
+                if|then|else|elseif|
+                do|while|
+                repeat|until|
+                for|from|to|downto|step|thru
```

Comment:
Please use the `words` function for this (and similarly below)

Reply:
I reworked the lexer to make use of `words` instead of alternation in the regex.